src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributeList.cs

```diff
@@ -78,106 +57,47 @@ public ReadOnlyTagHelperAttributeList(IEnumerable<TAttribute> attributes)
                 }
 
                 // Perf: Avoid allocating enumerator
-                for (var i = 0; i < Attributes.Count; i++)
+                for (var i = 0; i < Items.Count; i++)
                 {
-                    if (NameEquals(name, Attributes[i]))
+                    if (NameEquals(name, Items[i]))
                     {
-                        return Attributes[i];
+                        return Items[i];
                     }
                 }
 
-                return default(TAttribute);
-            }
-        }
-
-        /// <inheritdoc />
-        public int Count => Attributes.Count;
-
-        /// <summary>
-        /// Determines whether a <typeparamref name="TAttribute"/> matching <paramref name="item"/> exists in the
-        /// collection.
-        /// </summary>
-        /// <param name="item">The <typeparamref name="TAttribute"/> to locate.</param>
-        /// <returns>
-        /// <c>true</c> if an <typeparamref name="TAttribute"/> matching <paramref name="item"/> exists in the
-        /// collection; otherwise, <c>false</c>.
-        /// </returns>
-        /// <remarks>
-        /// <paramref name="item"/>s <see cref="IReadOnlyTagHelperAttribute.Name"/> is compared case-insensitively.
-        /// </remarks>
-        public bool Contains(TAttribute item)
-        {
-            if (item == null)
-            {
-                throw new ArgumentNullException(nameof(item));
+                return null;
             }
-
-            return Attributes.Contains(item);
         }
 
         /// <summary>
-        /// Determines whether a <typeparamref name="TAttribute"/> with the same
-        /// <see cref="IReadOnlyTagHelperAttribute.Name"/> exists in the collection.
+        /// Determines whether a <typeparamref name="TagHelperAttribute"/> with the same
+        /// <see cref="TagHelperAttribute.Name"/> exists in the collection.
```

Comment:
The `string` indexer uses better words: `with <see cref="TagHelperAttribute.Name"/> matching <paramref name="name"/>`.